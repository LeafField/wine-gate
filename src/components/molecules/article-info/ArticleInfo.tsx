import React, { FC } from "react";
import ArticleBalance from "../../atoms/article-balance/ArticleBalance";
import { TasteBalanceProps } from "../../../types/article_types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";

type Props = {
  red: boolean;
  favorite_count?: number;
} & TasteBalanceProps;

const ArticleInfo: FC<Props> = ({
  fruity,
  red,
  sober_or_sweet,
  tart,
  favorite_count,
}) => {
  return (
    <div className="flex w-fit gap-x-4">
      <ArticleBalance title={red ? "渋味" : "甘味"} balance={sober_or_sweet} />
      <ArticleBalance title="酸味" balance={tart} />
      <ArticleBalance title="果実味" balance={fruity} />
      {favorite_count && (
        <div className="flex items-center">
          <FontAwesomeIcon
            icon={faStar}
            className="text-sm text-yellow"
            data-testid="fa-star"
          />
          :<span className="text-main-text">{favorite_count}</span>
        </div>
      )}
    </div>
  );
};

export default ArticleInfo;
