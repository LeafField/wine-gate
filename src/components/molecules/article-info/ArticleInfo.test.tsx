import { screen, render } from "@testing-library/react";
import ArticleInfo from "./ArticleInfo";

describe("ArticleInfoの単体テスト", () => {
  test("ArticleInfoが正しく表示される", () => {
    const { container } = render(
      <ArticleInfo red={false} fruity={1} sober_or_sweet={2} tart={3} />,
    );
    expect(screen.getByText("果実味:1")).toBeInTheDocument();
    expect(screen.getByText("甘味:2")).toBeInTheDocument();
    expect(screen.getByText("酸味:3")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("red=trueの場合、渋味の項目が表示される", () => {
    render(<ArticleInfo red={true} fruity={4} sober_or_sweet={5} tart={3} />);
    expect(screen.getByText("渋味:5")).toBeInTheDocument();
  });

  test("favorite_countが存在する場合、お気に入り数が表示される", () => {
    render(
      <ArticleInfo
        red={false}
        fruity={4}
        sober_or_sweet={5}
        tart={3}
        favorite_count={10}
      />,
    );
    expect(screen.getByTestId("fa-star")).toBeInTheDocument();
    expect(screen.getByText("10")).toBeInTheDocument();
  });
});
