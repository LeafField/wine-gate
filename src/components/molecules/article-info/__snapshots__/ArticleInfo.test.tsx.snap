// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleInfoの単体テスト ArticleInfoが正しく表示される 1`] = `
<div>
  <div
    class="flex w-fit gap-x-4"
  >
    <p
      class="text-main-text"
    >
      甘味
      :
      2
    </p>
    <p
      class="text-main-text"
    >
      酸味
      :
      3
    </p>
    <p
      class="text-main-text"
    >
      果実味
      :
      1
    </p>
  </div>
</div>
`;
